### PR TITLE
feat(PPDSC-2798): expose css variables

### PR DESCRIPTION
**Pull request by mutebg**
PPDSC-2798

**What**

1. Background - why this is needed
2. What did you do - added CSS variables to theme provider
3. What does the reviewers should expect

**I have done:**
- [x] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [x] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by pp-serviceaccount** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2798-expose-css-variables
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2798-expose-css-variables/storybook

**Comment by LukeFinch** [comment]
All looks good, just a few questions:

- Do we want to namespace our variables to avoid potential clashes with other libraries?
- Have we considered data-attributes rather than class names?
  - something like data-theme="newskitLightTheme"

**Comment by mutebg** [comment]
> All looks good, just a few questions:
> 
> * Do we want to namespace our variables to avoid potential clashes with other libraries?
That's a good one, however how likely is The Sun to use other UI libraries at the same time?

> * Have we considered data-attributes rather than class names?
>   
>   * something like data-theme="newskitLightTheme"
Can you elaborate on that, I don't understand it

**Review by LukeFinch — DISMISSED** [review]

File: site/pages/_app.tsx [hunk 1]
```diff
@@ -156,6 +155,7 @@ export default class MyApp extends App<Props, State> {
 
         <NewsKitProvider
           theme={theme}
+          themeOptions={{exposeCSSVariables: true}}
```
Comment (JohnTParsons) [inline site/pages/_app.tsx:158]: Would `exposeCSSVariables` be better camelCase if called `exposeCssVariables`?

Comment (mutebg) [reply site/pages/_app.tsx:158]: I don't know really, from one side camelcase is more readable, on the other CSS is an acronym and as such, it needs to be upper-case.  Happy to change it if you think that's better

**Review by Xin00163 — COMMENTED** [review]
Nice work. Tested a few scenarios and all look good

File: src/theme/__tests__/theme-provider.test.tsx [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        1st level
+        <ThemeProvider theme={childrenTheme} exposeCSSVariables>
+          2nd level
+        </ThemeProvider>
+      </ThemeProvider>,
+    );
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  it.only('themeDiff -> different token', () => {
```
[the remark this answers is not in the document]
Comment (mutebg) [reply src/theme/__tests__/theme-provider.test.tsx]: Thanks, fixed

**Review by Xin00163 — APPROVED** [review]

File: site/pages/_app.tsx (hunk 1 above)
Comment (JohnTParsons) [reply site/pages/_app.tsx:158]: I'm inclined to think that exposeCssVariables is better

**Review by JohnTParsons — APPROVED** [review]

**Review by Xin00163 — APPROVED** [review]

**Comment by mstuartf** [comment]
I think @LukeFinch's point about namespacing is a good one. But I think it's fine for now. If it turns out to be something we need, we could add support for an optional namespace prefix prop to be passed in along with `exposeCssVariables`.

**Review by mstuartf — APPROVED** [review]

**Comment by mutebg** [comment]
> I think @LukeFinch's point about namespacing is a good one. But I think it's fine for now. If it turns out to be something we need, we could add support for an optional namespace prefix prop to be passed in along with `exposeCssVariables`.

Thanks, I totally agree, let's see if anyone needs that we can easily add it as an additional option to the theme provider.

**Review by Xin00163 — DISMISSED** [review]

**Review by LukeFinch — DISMISSED** [review]
Nice work 👏

**Review by JohnTParsons — DISMISSED** [review]

**Pull request merged**